### PR TITLE
added new logging monitoring

### PR DESCRIPTION
**Pull request by bulicamihai**

### CODE REVIEW

**Comment by bulicamihai** [comment]
[New Content Monitoring: output log lines for each event/app](https://trello.com/c/Bvwc5JRH/197-new-content-monitoring-output-log-lines-for-each-event-app)

File: consumer/consume.go [hunk 1]
```diff
@@ -53,43 +53,46 @@ func NewMessageQueueHandler(contentUriWhitelist *regexp.Regexp, contentTypeWhite
 
 func (qHandler *simpleMessageQueueHandler) HandleMessage(queueMsg kafka.FTMessage) error {
 	msg := NotificationQueueMessage{queueMsg}
-
+	contentType := msg.Headers["Content-Type"]
+	tid := msg.TransactionID()
 	pubEvent, err := msg.ToPublicationEvent()
+
+	log.WithMonitoringEvent("NotificationsPush", tid, contentType)
```
Comment (ioanaciont) [inline consumer/consume.go]: You're trying to use here `contentType`, but it only later gets checked if null or empty.

File: resources/history.go [hunk 2]
```diff
@@ -15,14 +15,14 @@ func History(history dispatch.History) func(w http.ResponseWriter, r *http.Reque
 
 		historyJSON, err := dispatch.MarshalNotificationsJSON(history.Notifications())
 		if err != nil {
-			log.WithError(err).Warn(errMsg)
+			log.WithField("event", "NotificationsPus").WithField("monitoring_event", "true").WithError(err).Warn(errMsg)
```
Comment (ioanaciont) [inline resources/history.go]: :pencil2: typo --> `NotificationsPus`

Comment (ioanaciont) [inline resources/history.go]: Also, use `log.WithMonitoringEvent`

File: resources/history.go [hunk 3]
```diff
@@ -15,14 +15,14 @@ func History(history dispatch.History) func(w http.ResponseWriter, r *http.Reque
...
 			http.Error(w, "", http.StatusInternalServerError)
 			return
 		}
 
 		_, err = w.Write(historyJSON)
 		if err != nil {
-			log.WithError(err).Warn(errMsg)
+			log.WithField("event", "NotificationsPush").WithField("monitoring_event", "true").WithError(err).Warn(errMsg)
```
Comment (ioanaciont) [inline resources/history.go]: use `log.WithMonitoringEvent`

File: resources/push.go [hunk 4]
```diff
@@ -54,7 +54,7 @@ func Push(reg dispatch.Registrar, apiGatewayKeyValidationURL string, httpClient
 
 		contentTypeParam, err := resolveContentType(r)
 		if err != nil {
-			log.WithError(err).Error("Invalid content type")
+			log.WithField("event", "NotificationsPush").WithField("monitoring_event", "true").WithError(err).Error("Invalid content type")
```
Comment (ioanaciont) [inline resources/push.go]: use `log.WithMonitoringEvent`

File: resources/push.go [hunk 5]
```diff
@@ -93,7 +93,9 @@ func Push(reg dispatch.Registrar, apiGatewayKeyValidationURL string, httpClient
 				logSuccessForHeartbeatMessage(notification, s)
 
 			case <-cn.CloseNotify():
-				log.WithField("subscriberId", s.Id()).
+				log.WithField("event", "NotificationsPush").
```
Comment (ioanaciont) [inline resources/push.go]: use `log.WithMonitoringEvent`

File: consumer/consume.go [hunk 6]
```diff
@@ -53,47 +53,45 @@ func NewMessageQueueHandler(contentUriWhitelist *regexp.Regexp, contentTypeWhite
 
 func (qHandler *simpleMessageQueueHandler) HandleMessage(queueMsg kafka.FTMessage) error {
 	msg := NotificationQueueMessage{queueMsg}
-	contentType := msg.Headers["Content-Type"]
 	tid := msg.TransactionID()
 	pubEvent, err := msg.ToPublicationEvent()
+	contentType := msg.Headers["Content-Type"]
 
-	log.WithMonitoringEvent("NotificationsPush", tid, contentType)
+	monitoringLogger :=log.WithMonitoringEvent("NotificationsPush", tid, contentType)
```
Comment (MihaiSorinMoisa) [inline consumer/consume.go]: Run `go fmt` for this line.

File: consumer/consume.go [hunk 7]
```diff
@@ -53,47 +53,45 @@ func NewMessageQueueHandler(contentUriWhitelist *regexp.Regexp, contentTypeWhite
...
 	notification, err := qHandler.mapper.MapNotification(pubEvent, msg.TransactionID())
 	if err != nil {
-		log.WithField("msg", string(msg.Body)).WithError(err).Warn("Skipping event: Cannot build notification for message.")
+		monitoringLogger.WithField("msg", string(msg.Body)).WithError(err).Warn("Skipping event: Cannot build notification for message.")
 		return err
 	}
-
-	log.WithField("resource", notification.APIURL).WithField("publish_reference", notification.PublishReference).Info("Valid notification received")
 	qHandler.dispatcher.Send(notification)
+	monitoringLogger.WithField("resource", notification.APIURL).Info("Valid notification received")
```
Comment (MihaiSorinMoisa) [inline consumer/consume.go]: This log should be without `monitoring_event`.

File: consumer/consume.go [hunk 8]
```diff
@@ -53,44 +53,45 @@ func NewMessageQueueHandler(contentUriWhitelist *regexp.Regexp, contentTypeWhite
...
 	msg := NotificationQueueMessage{queueMsg}
-
+	tid := msg.TransactionID()
 	pubEvent, err := msg.ToPublicationEvent()
+	contentType := msg.Headers["Content-Type"]
+
+	monitoringLogger :=log.WithMonitoringEvent("NotificationsPush", tid, contentType)
 	if err != nil {
-		log.WithField("transaction_id", msg.TransactionID()).WithField("msg", msg.Body).WithError(err).Warn("Skipping event.")
+		monitoringLogger.WithField("msg", msg.Body).WithError(err).Warn("Skipping event.")
```
Comment (emilolosutean) [inline consumer/consume.go]: .Warn overwrites msg field

File: dispatch/dispatcher_test.go [hunk 9]
```diff
@@ -36,8 +40,22 @@ var n2 = Notification{
 	PublishReference: "tid_test2",
 	LastModified:     "2016-11-02T10:55:24.244Z",
 }
+
+var n3 = Notification{
```
Comment (MihaiSorinMoisa) [inline dispatch/dispatcher_test.go]: This variable is unused.

File: dispatch/dispatcher_test.go [hunk 10]
```diff
@@ -320,6 +339,39 @@ func TestDispatchedNotificationsInHistory(t *testing.T) {
 	assert.NotContains(t, h.Notifications(), n1, "History does not contain old notification")
 }
 
+func TestShouldNotificationsToSubscribersFailed(t *testing.T) {
```
Comment (MihaiSorinMoisa) [inline dispatch/dispatcher_test.go]: This test doesn't actually work properly. The asserts aren't actually ever called.

File: dispatch/dispatcher_test.go [hunk 11]
```diff
@@ -320,6 +339,39 @@ func TestDispatchedNotificationsInHistory(t *testing.T) {
...
+
+	go d.Start()
+	defer d.Stop()
+
+	d.Register(s1)
+	d.Register(s2)
+	d.Register(s3)
+
+	d.Send(n1)
+	readFromSubscriberChannel(t, s1)
```
Comment (MihaiSorinMoisa) [inline dispatch/dispatcher_test.go]: What is the purpose of these read functions? No notifications will be sent to those channels.

File: dispatch/dispatcher_test.go [hunk 12]
```diff
@@ -320,6 +330,39 @@ func TestDispatchedNotificationsInHistory(t *testing.T) {
...
+	defer d.Stop()
+
+	d.Register(s1)
+	d.Register(s2)
+	d.Register(s3)
+
+	d.Send(n1)
+
+	time.Sleep(time.Second)
+    logger.Info("test")
```
Comment (MihaiSorinMoisa) [inline dispatch/dispatcher_test.go]: Remove this log line.

File: consumer/consume_test.go [hunk 13]
```diff
@@ -97,9 +97,10 @@ func TestAcceptNotificationBasedOnContentType(t *testing.T) {
 
 	handler := NewMessageQueueHandler(defaultContentUriWhitelist, contentTypeWhitelist, mapper, dispatcher)
 
-	msg := kafka.NewFTMessage(map[string]string{"X-Request-Id": "tid_summin", "Content-Type": "application/vnd.ft-upp-article+json"},
+	msg := kafka.NewFTMessage(map[string]string{"X-Request-Id": "tid_summin", "Content-Type": "application/vnd.ft-upp-article+json; version=1.0; charset=utf-8"},
 	`{"ContentURI": "http://not-in-the-whitelist.svc.ft.com:8080/lists/blah/55e40823-6804-4264-ac2f-b29e11bf756a"}`)
 
+
```
Comment (MihaiSorinMoisa) [inline consumer/consume_test.go]: Remove this empty line.

**Pull request merged**